pkg/kubectl/explain.go

```diff
@@ -0,0 +1,394 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/emicklei/go-restful/swagger"
+	"github.com/olekukonko/tablewriter"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	opMode    string = "operation"
+	modelMode string = "model"
+)
+
+var resource2Model = map[string]string{
```

Comment:
Ideally we'd use expandResourceShortcut to map short form to long form, and then map to model:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/kubectl.go#L93

I believe there should be a singularizer somewhere, also, that can be used. @JanetKuo